client/js/components/create_form.jsx

```diff
@@ -0,0 +1,258 @@
+var React = require("react");
+var _ = require("lodash");
+var $ = require("jquery");
+var router;
+var AppStore = require("../store/app_store");
+
+var CreateForm = React.createClass ({
+  componentWillMount: function() {
+    console.log("update func", this.forceUpdate);
+    router = this.props.router;
+  },
+  componentDidMount: function() {
+    AppStore.addChangeListener(this.forceUpdate);
+  },
+  componentWillUnmount: function() {
+    AppStore.removeChangeListener(this.forceUpdate);
+  },
+  trigger: function(){
+    this.forceUpdate();
+  },
+  createFormSchema: function(event){
+    event.preventDefault();
+    var schemaName = AppStore.formTitle.split(" ").join("_");
+    var URL = "/api/schema" + schemaName;
+    var DATA = {};
+    AppStore.questionsInEdit.map(function(elem){
+      DATA[elem[2]]=elem;
+    });
+    $.ajax({
+      url: URL,
+      type:"PUT",
+      data: DATA,
+      success: function(data) {
+        console.log("Sucess!! Data logged: ", data);
+        router.navigate("/", {trigger: true});
+        AppStore.emitChange();
+      },
+      error: function(error) {
+        console.error(error);
+      }
+    });
+  },
+  createNewQuestion: function(form){
+    console.log("in createNewQuestion2");
+    event.preventDefault();
+    var index = AppStore.questionsInEdit.length;
+    var questionObject = [];
+    questionObject.push(index);
+    questionObject.push("draggableQuestionInactive");
+    for(var i=0; i < form.target.length; i++){
+      questionObject.push(form.target[i].value);
+    }
+    console.log("QuestionObject: ", questionObject);
+    AppStore.questionsInEdit.push(questionObject);
+    console.log("questionsInEdit: ", AppStore.questionsInEdit);
+    this.trigger();
+  },
+  getSupportingFormFields: function() {
+    AppStore.currentQuestionType=[this.refs.questionType.getDOMNode().value];
+    this.trigger();
+  },
+  changeFormTitle: function(){
+    AppStore.formTitle = this.refs.formTitle.getDOMNode().value;
+    this.trigger();
+  },
+  outputUpdate: function(id) {
+    console.log("in outputUpdate function", "#scale"+id);
+            document.querySelector("#output"+id).value = this.refs["scale"+id].getDOMNode().value;
+  },
+  drop: function(questionIndex) {
+    event.preventDefault();
+    console.log("questionIndex in DROP", questionIndex);
+    var targetIndex = AppStore.questionDropTarget = questionIndex; 
+    var movingIndex = AppStore.questionBeingDragged;    
+    this.toggleQuestionActive(questionIndex);    
+    AppStore.questionsInEdit[AppStore.questionBeingDragged][0]=questionIndex;         
+    AppStore.questionsInEdit[targetIndex][0]=movingIndex;
+    var swap = AppStore.questionsInEdit[movingIndex];
```

Comment:
define all variables at the [top](http://javascript.crockford.com/code.html) of the function